mypy/nodes.py

```diff
@@ -1972,7 +1976,29 @@ class is generic then it will be a type constructor of higher kind.
     subtypes = None  # type: Set[TypeInfo] # Direct subclasses encountered so far
     names = None  # type: SymbolTable      # Names defined directly in this type
     is_abstract = False                    # Does the class have any abstract attributes?
+    is_protocol = False                    # Is this a protocol class?
+    runtime_protocol = False               # Does this protocol support isinstance checks?
     abstract_attributes = None  # type: List[str]
+    protocol_members = None  # type: List[str]
+
+    # These represent structural subtype matrices. Note that these are shared
```

Comment:
The first sentence is not descriptive enough. This is a pretty tricky concept so please explain it in more detail and maybe give an example.